File: freecodecamp/2026-04-21.py
import unittest

import logging

logging.basicConfig(level=logging.DEBUG, format="%(levelname)s: %(message)s")


def get_odd_words(args):
    logging.debug(f"start of get_odd_words {args=}")
    ret = []
    for w in args.split(" "):
        if len(w) % 2 == 1:
            ret.append(w)
    logging.debug(f"{ret=}")
    return " ".join(ret)


class TestCodeChallenge(unittest.TestCase):
    def test_daily_coding_challenge(self):
        self.assertEqual(get_odd_words("This is a super good test"), "a super")
        self.assertEqual(get_odd_words("one two three four"), "one two three")
        self.assertEqual(
            get_odd_words("banana split sundae with rainbow sprinkles on top"),
            "split rainbow sprinkles top",
        )
        self.assertEqual(
            get_odd_words("The quick brown fox jumped over the lazy river"),
            "The quick brown fox the river",
        )


if __name__ == "__main__":

    unittest.main(verbosity=2)

File: freecodecamp/2026-04-21.js
function getOddWords(args) {
  console.log(JSON.stringify(args));
  const ret = args.split(" ").filter((w) => w.length % 2 === 1);
  console.log(ret.join(" "));
  return ret.join(" ");
}
// Tests
const tests = [
  [getOddWords("This is a super good test"), "a super"],
  [getOddWords("one two three four"), "one two three"],
  [
    getOddWords("banana split sundae with rainbow sprinkles on top"),
    "split rainbow sprinkles top",
  ],
  [
    getOddWords("The quick brown fox jumped over the lazy river"),
    "The quick brown fox the river",
  ],
];

tests.forEach(([result, expected], i) => {
  const pass = JSON.stringify(result) === JSON.stringify(expected);
  // console.log(pass, JSON.stringify(result), JSON.stringify(expected));
  console.log(`Test ${i + 1}: ${pass ? "PASS" : "FAIL"}`);
});
